src/TextArea/TextArea.tsx

```diff
@@ -174,15 +174,19 @@ export class TextArea extends React.Component<Props, State> {
     };
 
     let rows = this.props.rows;
-    if (this.props.placeholder) {
+    if (this.props.placeholder && !this.props.readOnly) {
       // Need to add another row for autoGrow since it seems to collapse in a way that conflicts with the placeholder
       // margin
       rows = this.props.rows + 1;
     }
 
     let textarea = <textarea {...textAreaProps} rows={rows} />;
     if (this.props.autoResize) {
-      rows = this.props.rows + 1;
+      if (this.props.readOnly) {
+        rows = this.props.rows;
+      } else {
```

Comment:
We might want to add a comment explaining this behavior like the placeholder condition above. It also might be more clear to have it as a separate conditional instead of mixing it with the `placeholder` and `autoResize` logic, but I'll leave that decision up to you. 🙂

Reply:
In my new commit I simplified the logic for setting rows in one conditional statement which makes it more readable.